src/App.tsx

```diff
@@ -444,15 +449,31 @@ function AppContent() {
               {getNurseriesView()}
             </Route>
             <Route exact path={APP_PATHS.INVENTORY}>
-              <Inventory hasNurseries={selectedOrgHasNurseries()} hasSpecies={selectedOrgHasSpecies()} />
+              {nurseryV2
+                ? <InventoryV2 hasNurseries={selectedOrgHasNurseries()} hasSpecies={selectedOrgHasSpecies()} />
+                : <Inventory hasNurseries={selectedOrgHasNurseries()} hasSpecies={selectedOrgHasSpecies()} />
+              }
             </Route>
             <Route exact path={APP_PATHS.INVENTORY_NEW}>
               <InventoryCreate />
             </Route>
             <Route path={APP_PATHS.INVENTORY_WITHDRAW}>
               <SpeciesBulkWithdrawWrapperComponent withdrawalCreatedCallback={() => setWithdrawalCreated(true)} />
             </Route>
-            <Route path={APP_PATHS.INVENTORY_ITEM}>
+            {nurseryV2 &&
+              <>
+                <Route path={APP_PATHS.INVENTORY_BATCH_FOR_NURSERY}>
+                  <InventoryBatch origin='Nursery' />
+                </Route>
+                <Route path={APP_PATHS.INVENTORY_BATCH_FOR_SPECIES}>
+                  <InventoryBatch origin='Species' />
+                </Route>
+                <Route path={APP_PATHS.INVENTORY_ITEM_FOR_NURSERY}>
+                  <InventoryViewForNursery />
+                </Route>
+              </>
```

Comment:
Thoughts on routing these inside another file/component?
Basically have a route handler for `/inventory` here, which instantiates a root component `<XYZ {props} />`
Under XYZ.ts define routing for the various inventory sub routes.
Will help keep App.ts routing more manageable.

Reply:
Yeah I can look at that. Thanks, Karthik!

Reply:
Was having some difficulty here, will defer

Reply:
No worries. Thanks for trying. For reference later see Observations.